### ISSUE TITLE
wrong icon for mp4 torrent

### ISSUE THREAD

**Issue by chantzish** [transmission/transmission#5965]
### What is the issue?

Torrents of mp4 files are being reported with primary-mime-type of "application/mp4" instead of "video/mp4".
This causes the toorent icon to be the application icon instead of video icon.

Note: mp4 and mpg4 were added as "application/mp4" above a line  defining them as "video/mp4" in https://github.com/transmission/transmission/commit/d2d30c273482583b812cadf32361e490fedd3b6a I guess the search stops on the first match

The icons of application and video are attached:

![image](https://github.com/transmission/transmission/assets/7995464/83e28cc9-5760-4648-a875-199b252cb065)

### Which application of Transmission?

Web client

### Which version of Transmission?

Transmission 4.1.0-dev 78a5d4da74

**Comment by killemov** [comment]
Weird that you attach these icons as the problem is clearly just a wrong `primary-mime-type`.

**Comment by chantzish** [comment]
I just meant the mime type has an effect on the visuals and I was expecting to see the video icon for mp4 files. The cause is indeed the returned `primary-mime-type`.

**Comment by ckerr** [comment]
This is coming from upstream at https://github.com/jshttp/mime-db/commit/48e411522aed9925570b6c6bbf0bbc331242c038 which ultimately is getting it from https://www.rfc-editor.org/rfc/rfc4337

tldr is that the correct mime-type depends on the codecs used in the file, which may not actually be available at the point where this code is called, e.g. maybe the file isn't even downloaded yet.

_Probably_ we should just hardcode a special case into `tr_get_mime_type_for_filename()` for this situation to default to `video/mp4` when more information isn't available, since that's the most common use case for torrents. IMO that's not completely satisfying, but I can't think of a more deterministic fix for the case where the file isn't downloaded yet. :man_shrugging:

**Comment by killemov** [comment]
It irks to coerce a container format to video, but it is the most common usage. (Who uses mp4 audio nowadays?) Maybe a "container/mp4" mime-type should exist. See https://developer.mozilla.org/en-US/docs/Web/Media/Formats/codecs_parameter and perhaps a tiny probe into the file could determine the actual mime-type.

**Comment by ckerr** [comment]
Fixed by #8377.

**Comment by ckerr** [comment]
Whoops let's wait for it to be merged first 😆
